File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        # Initialize variables
        max_sum = current_sum = nums[0]
        
        # Iterate through the array starting from the second element
        for num in nums[1:]:
            # Update current_sum to either the current number itself or current_sum + num
            current_sum = max(num, current_sum + num)
            
            # Update max_sum to the maximum of max_sum or current_sum
            max_sum = max(max_sum, current_sum)
        
        return max_sum

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int size = nums.size();
        int sum = 0;
        int maxi = INT_MIN;

        for(int i = 0; i < size; i++)
        {
            sum += nums[i];

            if(sum > maxi)
            {
                maxi = sum;
            }

            if(sum < 0)
            {
                sum = 0;
            }
        }
        return maxi;

    }
};